common/gps.h

```diff
@@ -0,0 +1,47 @@
+//GPS
+
+
+#include "nmea_parser.h"
+#include "serial_reader.h"
+#include "../third_party/units.h"
+#include "map_coordinate.h"
+
+namespace BoBRobotics 
+{
+namespace GPS 
+{
+
+class Gps {
+
+    private:
+    
+    SerialReader sreader;
+    const char *m_device_path;
+    
+    public:
+
+    Gps(const char *device_path) : m_device_path(device_path) {
+
+        try {
+            sreader.connect(device_path);
+            if (!sreader.isConnected()) throw "device not found";
+            m_device_path = device_path;
+        } catch(const char* error) {
+            std::cout << "[Gps:Gps()]: " << error <<  std::endl;
+            throw error;
+        }
+    }
+
+    GPSData getGPSData() {
+        GPSData data;
+        try {
+            data = NMEAParser::parseNMEA(sreader.readData()); 
+            return data;
+        } catch(const char *s) {
+            std::cout << "[Gps:getGPSPosition]: " << s <<  std::endl;
```

Comment:
Again, I'd just let it pass up the call stack.

As a side note, for this kind of printing you can use plog:
```c++
LOG_ERROR << s;
```
It prints the function name and line number etc. automatically.